code/modules/research/xenobiology/crossbreeding/warping

```diff
@@ -0,0 +1,796 @@
+			qdel(C)
+
+
+
+
+
+
+/* the blue warp rune  keeps a tile slippery CONSTANTLY by adding water (maybe I'll make it lube if I feel extra fucky) every few(10) seconds */
+
+
+
+/obj/item/slimecross/warping/blue
+	colour = "blue"
+	runepath = /obj/effect/warped_rune/cyanspace //we'll call the blue rune cyanspace to not mix it up with actual bluespace rune
+	effect_desc = "creates a rune that constantly wet itself with water as long as the rune is up"
+
+/obj/effect/warped_rune/cyanspace
+	icon_state = "slipperyspace"
+	desc = "Water drowns, water feeds, water slips, is there a thing water cannot do?"
+
+/obj/effect/warped_rune/cyanspace/Initialize()
+	.=..()
+	START_PROCESSING(SSprocessing, src)
+
+/obj/effect/warped_rune/cyanspace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		slippery_rune(T)
+
+
+///just wet the turf under the rune every now and then, there's actually a small moment where the tile isn't wet
+/obj/effect/warped_rune/cyanspace/proc/slippery_rune(turf/open/T)
+	T.MakeSlippery(TURF_WET_WATER, min_wet_time = 10 SECONDS, wet_time_to_add = 5 SECONDS)
+
+
+
+
+
+/*      Metal rune : makes an invisible wall. actually I lied, the rune is the wall.*/
+
+
+/obj/item/slimecross/warping/metal
+	colour = "metal"
+	runepath = /obj/effect/warped_rune/metalspace
+	effect_desc = "Creates a rune that prevents passage above it, takes longer to store and draw than other runes."
+	storing_time = 200  //storing time is longer for this particular rune
+	drawing_time = 100  //it's also longer to draw it
+
+//It's a wall what do you want from me
+/obj/effect/warped_rune/metalspace
+	desc = "Words are powerful things, they can stop someone dead in their tracks if used in the right way"
+	icon_state = "metal_space"
+	density = TRUE
+
+
+
+
+
+
+
+
+/*  Yellow rune space acts as an infinite generator, works without power and anywhere, recharges the APC of the room it's in and any batteries put on it
+
+*/
+
+
+/obj/item/slimecross/warping/yellow
+	colour = "yellow"
+	runepath = /obj/effect/warped_rune/yellowspace
+	effect_desc = "Creates a rune that Infinitely recharge batteries that are put on it, it will also passively recharge the APC of the room"
+
+
+/obj/effect/warped_rune/yellowspace
+	desc = "Where does all this energy come from? Who knows,the process does not matter, only the result."
+	icon_state = "elec_rune"
+
+
+/obj/effect/warped_rune/yellowspace/Initialize()
+	.=..()
+	START_PROCESSING(SSmachines, src)
+
+
+/obj/effect/warped_rune/yellowspace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		apc_charge()
+		electrivore()
+
+/obj/effect/warped_rune/yellowspace/proc/apc_charge()
+
+	var/area/A = get_area(T)
+	for(var/obj/machinery/power/apc/P in A)
+		if(!P.cell)
+			return
+		if(P.cell.charge <! (P.cell.maxcharge - 1))
+			return
+		P.cell.charge += P.cell.maxcharge/100 //will basically recharge 1% of the APC cell every 10 seconds,we don't want to run engineers out of buisness
```

Comment:
this APC-recharging stuff needs a buff in recharging speed. remember that you could have just slapped a yellow slime cell into an APC to make that APC basically unable to run out of power anyway, so the only advantage that making a rune out of your yellow slime instead of a yellow slime cell would have is the ability to recharge other cells and the portability/ease of setting it up, and the yellow slime cell is 10 bluespace slime cores cheaper than this rune is